lib/smart_proxy_openscap/spool_forwarder.rb

```diff
@@ -51,19 +43,28 @@ def forward_date_dir(cname, policy_id, date, date_dir)
           forward_arf_file(cname, policy_id, date, arf_path)
         end
       end
-      remove(date_dir)
+      remove_if_empty(date_dir)
     end
 
     def forward_arf_file(cname, policy_id, date, arf_file_path)
       data = File.open(arf_file_path, 'rb') { |io| io.read }
       post_to_foreman = ForemanForwarder.new.post_arf_report(cname, policy_id, date, data)
       Proxy::OpenSCAP::StorageFS.new(Proxy::OpenSCAP::Plugin.settings.reportsdir, cname, post_to_foreman['id'], date).store_archive(data)
       File.delete arf_file_path
+    rescue OpenSCAP::OpenSCAPError => e
+      logger.error "Failed to parse Arf Report at #{arf_file_path}, moving to #{Proxy::OpenSCAP::Plugin.settings.corrupted_dir}"
+
+      Proxy::OpenSCAP::StorageFS.new(Proxy::OpenSCAP::Plugin.settings.corrupted_dir, cname, policy_id, date)
+                                .move_corrupted(arf_file_path.split('/').last)
+      # TODO: call Foreman? What if server is down? Do we store warnings and retry?
```

Comment:
Yeah, already addressed on the Foreman side by going through [proxy logs](https://github.com/theforeman/foreman_openscap/pull/235)